flink-core/src/main/java/org/apache/flink/api/common/typeutils/TypeSerializerConfigSnapshot.java

```diff
@@ -155,4 +158,38 @@ else if (serializer instanceof UnloadableDummyTypeSerializer) {
 				TypeSerializerSchemaCompatibility.incompatible() :
 				TypeSerializerSchemaCompatibility.compatibleAsIs();
 	}
+
+	/**
+	 * SelfResolving assists with the migration path to the new serialization abstraction.
+	 *
+	 * <p>Previously some {@link TypeSerializer}s were reusing the same {@link TypeSerializerConfigSnapshot} class in
```

Comment:
For this paragraph, I would suggest to simply imply that it can be used for general cases where the `ensureCompatibility` method cannot be removed. The fact that some of our serializers were sharing the same `TypeSerializerConfigSnapshot` class, as mentioned here, is only one of the possible reasons.
Another reason, as we discussed before, is that users were using a Flink-shipped serializer before and does not have access to the serializer class in user code.

Reply:
Good idea, changing.